src/jit/lclvars.cpp

```diff
@@ -1833,7 +1833,8 @@ bool Compiler::lvaShouldPromoteStructVar(unsigned lclNum, lvaStructPromotionInfo
 
             // TODO-PERF - Implement struct promotion for incoming multireg structs
             //             Currently it hits assert(lvFieldCnt==1) in lclvar.cpp line 4417
-
+            //             Also there will be a problem with byte moves, because now it uses 4byte move.
+            //             It could rewrite the other fields in a promoted struct.
```

Comment:
You should be more specific here and say that it is the implementation of jmp that uses the 4 byte move to store byte parameters to the stack, so that if we have a byte field with something else occupying the same 4-byte slot, it will overwrite other fields.